cura/Settings/MachineManager.py

```diff
@@ -847,6 +847,24 @@ def variantBuildplateUsable(self) -> bool:
 
         return result
 
+    @pyqtProperty(bool, notify = activeMaterialChanged)
+    def variantCoreUsableForFactor4(self) -> bool:
+        """The selected core is usable if it is in second extruder of Factor4
+        """
+        result = True
+        if not self._global_container_stack:
+            return result
+        if self.activeMachine.id != "UltiMaker Factor 4":
+            return result
+
+        for extruder_container in self._global_container_stack.extruderList:
+            if( extruder_container.id.startswith("ultimaker_factor4_extruder_right")):
```

Comment:
also you should use the definition id
```suggestion
            if extruder_container.definition.id.startswith("ultimaker_factor4_extruder_right"):
```